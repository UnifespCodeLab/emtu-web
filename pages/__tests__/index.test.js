import { mount, createLocalVue, RouterLinkStub } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import VueRouter from 'vue-router'

import SearchPage from '~/pages/index.vue'
RouterLinkStub

describe('Pages / SearchPage', () => {
  const defaultData = {
    cities: ['São José dos Campos', 'Jacareí', 'Taubaté'],
    date: new Date(Date.now() - new Date().getTimezoneOffset() * 60000)
      .toISOString()
      .substr(0, 10),
    modalDate: false,
    time: `${new Date().getHours()}:${new Date().getMinutes()}`,
    modalTime: false
  }

  const localVue = createLocalVue()
  let vuetify
  let router

  beforeEach(() => {
    vuetify = new Vuetify()
    router = new VueRouter()
  })

  describe('when all data is correct', () => {
    it('should render the page', () => {
      const wrapper = mount(SearchPage, {
        localVue,
        vuetify,
        router,
        components: {
          RouterLinkStub
        },
        data () {
          return defaultData
        }
      })
      expect(wrapper.findAll('input').length).toBe(8)
      expect(wrapper.find('.search-page__image').exists()).toBe(true)
      expect(wrapper.element).toMatchSnapshot()
    })
  })
})
