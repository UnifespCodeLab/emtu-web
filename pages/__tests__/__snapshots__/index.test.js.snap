// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pages / SearchPage when all data is correct should render the page 1`] = `
<div
  class="search-page"
>
  <div
    class="search-page__form"
  >
    <div
      class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select v-autocomplete"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-4"
          class="v-input__slot"
          role="combobox"
        >
          <div
            class="v-select__slot"
          >
            <label
              class="v-label theme--light"
              for="input-4"
              style="left: 0px; position: absolute;"
            >
              Origem
            </label>
            <input
              autocomplete="off"
              id="input-4"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-menu-down theme--light"
                />
              </div>
            </div>
            <input
              type="hidden"
            />
          </div>
          <div
            class="v-menu"
          >
            <!---->
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select v-autocomplete"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-9"
          class="v-input__slot"
          role="combobox"
        >
          <div
            class="v-select__slot"
          >
            <label
              class="v-label theme--light"
              for="input-9"
              style="left: 0px; position: absolute;"
            >
              Destino
            </label>
            <input
              autocomplete="off"
              id="input-9"
              type="text"
            />
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-menu-down theme--light"
                />
              </div>
            </div>
            <input
              type="hidden"
            />
          </div>
          <div
            class="v-menu"
          >
            <!---->
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="v-input theme--light v-text-field v-text-field--single-line v-text-field--solo v-text-field--enclosed v-select"
    >
      <div
        class="v-input__control"
      >
        <div
          aria-expanded="false"
          aria-haspopup="listbox"
          aria-owns="list-14"
          class="v-input__slot"
          role="button"
        >
          <div
            class="v-select__slot"
          >
            <label
              class="v-label theme--light"
              for="input-14"
              style="left: 0px; position: absolute;"
            >
              Cid
            </label>
            <div
              class="v-select__selections"
            >
              <input
                aria-readonly="false"
                autocomplete="off"
                id="input-14"
                readonly="readonly"
                type="text"
              />
            </div>
            <div
              class="v-input__append-inner"
            >
              <div
                class="v-input__icon v-input__icon--append"
              >
                <i
                  aria-hidden="true"
                  class="v-icon notranslate mdi mdi-menu-down theme--light"
                />
              </div>
            </div>
            <input
              type="hidden"
            />
          </div>
          <div
            class="v-menu"
          >
            <!---->
          </div>
        </div>
        <div
          class="v-text-field__details"
        >
          <div
            class="v-messages theme--light"
          >
            <transition-group-stub
              class="v-messages__wrapper"
              name="message-transition"
              tag="div"
            />
          </div>
        </div>
      </div>
    </div>
     
    <div
      class="search-page__time-container"
    >
      <div
        class="v-input v-input--is-label-active v-input--is-dirty v-input--is-readonly theme--light v-text-field"
      >
        <div
          class="v-input__prepend-outer"
        >
          <div
            class="v-input__icon v-input__icon--prepend"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-clock-horaViagem-four-outline theme--light"
            />
          </div>
        </div>
        <div
          class="v-input__control"
        >
          <div
            class="v-input__slot"
          >
            <div
              class="v-text-field__slot"
            >
              <label
                class="v-label v-label--active theme--light"
                for="input-20"
                style="left: 0px; position: absolute;"
              >
                Selecione o hor√°rio
              </label>
              <input
                aria-expanded="false"
                aria-haspopup="true"
                id="input-20"
                readonly="readonly"
                role="button"
                type="text"
              />
            </div>
          </div>
          <div
            class="v-text-field__details"
          >
            <div
              class="v-messages theme--light"
            >
              <transition-group-stub
                class="v-messages__wrapper"
                name="message-transition"
                tag="div"
              />
            </div>
          </div>
        </div>
      </div>
      <div
        class="v-dialog__container"
      >
        <!---->
      </div>
       
      <div
        class="v-dialog__container"
      >
        <!---->
      </div>
      <div
        class="v-input v-input--is-label-active v-input--is-dirty v-input--is-readonly theme--light v-text-field"
      >
        <div
          class="v-input__prepend-outer"
        >
          <div
            class="v-input__icon v-input__icon--prepend"
          >
            <i
              aria-hidden="true"
              class="v-icon notranslate mdi mdi-calendar theme--light"
            />
          </div>
        </div>
        <div
          class="v-input__control"
        >
          <div
            class="v-input__slot"
          >
            <div
              class="v-text-field__slot"
            >
              <label
                class="v-label v-label--active theme--light"
                for="input-25"
                style="left: 0px; position: absolute;"
              >
                Selecione a data
              </label>
              <input
                aria-expanded="false"
                aria-haspopup="true"
                id="input-25"
                readonly="readonly"
                role="button"
                type="text"
              />
            </div>
          </div>
          <div
            class="v-text-field__details"
          >
            <div
              class="v-messages theme--light"
            >
              <transition-group-stub
                class="v-messages__wrapper"
                name="message-transition"
                tag="div"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
     
    <button
      class="v-btn v-btn--block v-btn--is-elevated v-btn--has-bg theme--light elevation-2 v-size--large primary"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        
      BUSCAR
    
      </span>
    </button>
  </div>
   
  <img
    alt="woman-hitchhiking"
    class="search-page__image"
    src="~/assets/images/woman-hitchhiking.jpg"
  />
</div>
`;
